athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/data/BlockTest.java

```diff
@@ -865,6 +866,117 @@ public void ListOfStructsTest()
+         * Generate and write the schema
+         */
+        SchemaBuilder schemaBuilder = new SchemaBuilder();
+        schemaBuilder.addField(
+                FieldBuilder.newBuilder("innerStruct", Types.MinorType.STRUCT.getType())
+                        .addStringField("varchar")
+                        .addListField("list", Types.MinorType.VARCHAR.getType())
+                        .build());
+        Schema origSchema = schemaBuilder.build();
+
+        /**
+         * Generate and write the block
+         */
+        Block expectedBlock = expectedAllocator.createBlock(origSchema);
+
+        int expectedRows = 200;
+        for (Field next : origSchema.getFields()) {
+            ValueVector vector = expectedBlock.getFieldVector(next.getName());
+            for (int i = 0; i < expectedRows; i++) {
+                switch (vector.getMinorType()) {
+                    case STRUCT:
+                        Map<String, Object> value = new HashMap<>();
+                        value.put("varchar", "chars");
+                        if (i % 2 == 0) {
+                            List<String> listVal = new ArrayList<>();
+                            listVal.add("value_0_" + i);
+                            listVal.add("value_1_" + i);
+                            value.put("list", listVal);
+                        }
+                        else {
+                            value.put("list", null);
+                        }
+                        BlockUtils.setComplexValue((StructVector) vector, i, FieldResolver.DEFAULT, value);
+                        break;
+                    default:
+                        throw new UnsupportedDataTypeException(vector.getMinorType() + " is not supported");
+                }
+            }
+        }
+        expectedBlock.setRowCount(expectedRows);
+
+        RecordBatchSerDe expectSerDe = new RecordBatchSerDe(expectedAllocator);
+        ByteArrayOutputStream blockOut = new ByteArrayOutputStream();
+        ArrowRecordBatch expectedBatch = expectedBlock.getRecordBatch();
+        expectSerDe.serialize(expectedBatch, blockOut);
+
+        assertSerializationOverhead(blockOut);
+        expectedBatch.close();
+        expectedBlock.close();
+
+        ByteArrayOutputStream schemaOut = new ByteArrayOutputStream();
+        SchemaSerDe schemaSerDe = new SchemaSerDe();
+        schemaSerDe.serialize(origSchema, schemaOut);
+        Schema actualSchema = schemaSerDe.deserialize(new ByteArrayInputStream(schemaOut.toByteArray()));
+
+        BlockAllocatorImpl actualAllocator = new BlockAllocatorImpl();
+        RecordBatchSerDe actualSerDe = new RecordBatchSerDe(actualAllocator);
+        ArrowRecordBatch batch = actualSerDe.deserialize(blockOut.toByteArray());
+
+        /**
+         * Generate and write the block
+         */
+        Block actualBlock = actualAllocator.createBlock(actualSchema);
+        actualBlock.loadRecordBatch(batch);
+        batch.close();
+
+        for (
```

Comment:
nit put in one line

Reply:
the for loop

Reply:
yikes, auto-formatter 0wned me. Fixed now